src/setcontest.c

```diff
@@ -224,7 +224,9 @@ void setcontest(char *name) {
     w_cty = getctynr(wcall);
     ve_cty = getctynr(vecall);
 
-    strcpy(whichcontest, name);
+    if (strcmp(whichcontest, name) != 0) {    /* remember name if new */
```

Comment:
> not sure why this is an issue, though.

No, it seems not to be an issue to copy to yourself. I just tried to get rid of the compiler warning about the overlapping memory.

> to avoid the situation when name is whichcontest, wouldn't
> if (name != whichcontest) be more appropriate?

To be honest, I sat there quite a while thinking what is the correct way. Technically (related to the warning) your test would be enough. I feared that someone could change whichcontest on the fly  so that a copy would be needed. Thinking again now I see that it can not happen as we hand over the address of the buffer to setcontest() not the string inside - so the simple compare of pointers would be enough. I will fix it.